File: rust/src/bubble_sort.rs
// Code originally forked from https://github.com/diptangsu/Sorting-Algorithms/
include!("constants.rs");
use std::time::Instant;

pub fn bubble_sort<T: Ord>(array: &mut [T]) {
    for i in 0..array.len() {
        for j in 0..array.len() - 1 - i {
            if array[j] > array[j + 1] {
                array.swap(j, j + 1);
            }
        }
    }
}

fn main() {
    unsafe {
        for _x in 0..50 {
            let start = Instant::now();
            bubble_sort(&mut SET1);
            let elapsed = start.elapsed();
            println!("{:?}", elapsed);
        }
    }
}

File: python/bubble_sort.py
# Code originally forked from https://github.com/diptangsu/Sorting-Algorithms/

import time
from constants import *


def bubble_sort(array):
    n = len(array)
    for i in range(n):
        swapped = False
        for j in range(0, n - i - 1):
            if array[j] > array[j + 1]:
                array[j], array[j + 1] = array[j + 1], array[j]
                swapped = True
        if not swapped:
            break


if __name__ == '__main__':
    for z in range(50):
        set_array = set2()
        start = time.time() * 1000
        bubble_sort(set_array)
        end = time.time() * 1000
        elapsed = end - start
        print(elapsed)
